### ISSUE TITLE
[BUG] Hauler fileserve fails on large files due to hardcoded 15 second timeout.

### ISSUE THREAD

**Issue by atoy3731** [hauler-dev/hauler#243]
**Environmental Info:**

```
Linux REDACTED 4.4.302+ #69057 SMP Fri Jan 12 17:02:27 CST 2024 x86_64 GNU/Linux synology_v1000_1821+
```

**Hauler Version:**

```
GitVersion:    1.0.3
GitCommit:     f7c24f6
GitTreeState:  clean
BuildDate:     2024-04-24T16:50:33Z
GoVersion:     go1.21.9
Compiler:      gc
Platform:      linux/amd64
```

**Describe the Bug:**

- When trying to pull a large file from the fileserver, the hardcoded timeout can affect the ability to pull that artifact.

See: https://github.com/rancherfederal/hauler/blob/main/internal/server/file.go#L36-L37

**Steps to Reproduce:**

- For me, I was utilizing `hauler store serve fileserver` to serve the Harvester ISO for PXE booting (5.6Gi in size).
- When pointing Harvester PXE boot to Hauler, the download of the ISO continuously fails at 15 seconds due to the timeout.

**Expected Behavior:**

One of two options:
- The timeout is configurable.
- The timeout is set to a higher amount than 15 seconds. Perhaps 60 or 120.

**Actual Behavior:**

- See above.

**Additional Context:**

- N/A

**Comment by KaminFay** [comment]
Just wanted to chime in here, I was really curious about this issue as I also hit it somewhat recently so just wanted to provide a second round of proof. 

## Environmental Info:

```
Linux e851a7a7b747 6.4.16-linuxkit #1 SMP PREEMPT Thu Nov 16 10:49:20 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux
```

## Hauler Version:

GitVersion:    1.0.3
GitCommit:     f7c24f6
GitTreeState:  clean
BuildDate:     2024-04-24T16:50:33Z
GoVersion:     go1.21.9
Compiler:      gc
Platform:      linux/amd64

## Describe the Bug:

When trying to pull a large file from the fileserver, the hardcoded timeout can affect the ability to pull that artifact.
See: https://github.com/rancherfederal/hauler/blob/main/internal/server/file.go#L36-L37

## Steps to Reproduce:

- Generated a 10Gb file:
```
dd if=/dev/urandom of=random.txt count=1024 bs=10M
tar -cf archive.tar random.txt
```

- Added it to the store:
```
root@e851a7a7b747:/# ./hauler store add file archive.tar
2024-05-26 23:04:09 INF adding 'file' [archive.tar] to the store as [hauler/archive.tar:latest]

root@e851a7a7b747:/# ./hauler store info
+-----------------------------------------------+------+----------+----------+---------+
| REFERENCE                                     | TYPE | PLATFORM | # LAYERS | SIZE    |
+-----------------------------------------------+------+----------+----------+---------+
| hauler/archive.tar:latest                     | file | -        |        1 | 10.7 GB |
| hauler/hauler_1.0.3_linux_amd64.tar.gz:latest | file | -        |        1 | 63.3 MB |
+-----------------------------------------------+------+----------+----------+---------+
|                                                                    TOTAL   | 10.8 GB |
+-----------------------------------------------+------+----------+----------+---------+
```
 - Waited....
 - Attempted to curl and would you look and I also saw the connection close at 15 seconds. 
```
PS C:\Users\user> curl.exe --verbose *.*.*.*:8080/archive.tar -o archive.tar
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying *.*.*.*:8080...
* Connected to *.*.*.* (*.*.*.*) port 8080
> GET /archive.tar HTTP/1.1
> Host: *.*.*.*:8080
> User-Agent: curl/8.4.0
> Accept: */*
>
< HTTP/1.1 200 OK
< Accept-Ranges: bytes
< Content-Length: 10737428480
< Content-Type: application/octet-stream
< Last-Modified: Sun, 26 May 2024 23:15:10 GMT
< Date: Sun, 26 May 2024 23:15:52 GMT
<
{ [512 bytes data]
  4 10.0G    4  431M    0     0  29.3M      0  0:05:48  0:00:14  0:05:34 30.1M* transfer closed with 10270052088 bytes remaining to read
  4 10.0G    4  445M    0     0  29.1M      0  0:05:50  0:00:15  0:05:35 29.3M
* Closing connection
curl: (18) transfer closed with 10270052088 bytes remaining to read
```

Expected Behavior:

A configurable timeout flag for fileservers.

Additional Context:

- I did attempt to try this with small and large files using curls `--limit-rate` flag but that keeps the connection alive and refreshed.

```
PS C:\Users\user> curl.exe *.*.*.*:8080/archive.tar -o archive.tar --limit-rate 1k
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying *.*.*.*:8080...
  0 10.0G    0 36352    0     0   1022      0 121d 14h  0:00:35 121d 14h  1019
```

**Comment by KaminFay** [comment]
Opened a PR to adjust this: #247 
As well as a documentation PR: [19](https://github.com/rancherfederal/hauler-docs/pull/19)

With the new timeout flag set to 1 it will auto close out:
```
~$ curl ****:8080/goland-2023.2.5-aarch64.dmg -o testing.dmg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  4  615M    4 30.5M    0     0  27.7M      0  0:00:22  0:00:01  0:00:21 27.7M
curl: (18) transfer closed with 613475259 bytes remaining to read
```
Setting it to 120 seconds results in everything pulling as expected.
```
~$ curl ****:8080/goland-2023.2.5-aarch64.dmg -o testing.dmg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  615M  100  615M    0     0  32.4M      0  0:00:18  0:00:18 --:--:-- 32.4M
```

**Comment by zackbradys** [comment]
Closing as resolved per PR https://github.com/rancherfederal/hauler/pull/247!
